crypto/mlkem/mlkem768.c

```diff
@@ -1194,4 +1169,100 @@ int ossl_mlkem768_decap(uint8_t *out_shared_secret,
     return mlkem_decap(out_shared_secret, ciphertext, priv, mlkem_ctx);
 }
 
+int ossl_mlkem768_marshal_public_key(uint8_t *out,
+                                     const struct ossl_mlkem768_public_key *public_key)
+{
+    struct public_key_RANK768 *pub = public_key_768_from_external(public_key);
+
+    return mlkem_marshal_public_key(out, pub);
+}
+
+/*
+ * mlkem_parse_public_key_no_hash parses |in| into |pub| but doesn't calculate
+ * the value of |pub->public_key_hash|.
+ */
+static int mlkem_parse_public_key_no_hash(public_key_RANK768 *pub, uint8_t *in,
+                                          ossl_mlkem_ctx *mlkem_ctx)
+{
+    if (vector_decode(&pub->t, in, kLog2Prime) != 1)
+        return 0;
+    memcpy(pub->rho, in+encoded_vector_size(RANK768), sizeof(pub->rho));
+    matrix_expand(&pub->m, pub->rho, mlkem_ctx);
+    return 1;
+}
+
+static int mlkem_parse_public_key(public_key_RANK768 *pub, uint8_t *in,
+                                  ossl_mlkem_ctx *mlkem_ctx)
+{
+    if (!mlkem_parse_public_key_no_hash(pub, in, mlkem_ctx))
+        return 0;
+    hash_h(pub->public_key_hash, in, OSSL_MLKEM768_PUBLIC_KEY_BYTES, mlkem_ctx);
+    return 1;
+}
+
+int ossl_mlkem768_parse_public_key(struct ossl_mlkem768_public_key *public_key,
+                                   uint8_t *in, ossl_mlkem_ctx *mlkem_ctx)
+{
+    struct public_key_RANK768 *pub = public_key_768_from_external(public_key);
+
+    return mlkem_parse_public_key(pub, in, mlkem_ctx);
```

Comment:
prudent to check pub != NULL?

Reply:
All the `_from_external` functions just do a cast, so I don't think that's needed